main.rb

```diff
@@ -0,0 +1,111 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    array.length.times { |index| yield(array[index]) }
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    array.length.times { |index| yield(array[index], index) }
+    self
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    array = []
+    my_each { |index| array.push(index) if yield(index) }
+    array
+  end
+
+  def my_all?(*arguments)
+    if !arguments[0].nil?
+      my_each { |index| return true unless arguments[0] == index }
+    elsif !block_given?
+      my_each { |index| return true unless index }
+    elsif arguments[0].is_a? Class
+      my_each { |index| return true unless index.class.ancestors.include?(arguments[0]) }
+    elsif arguments[0].is_a? Regexp
+      my_each { |index| return true unless arguments[0].match(index) }
+    else
+      my_each { |index| return true unless yield(index) }
+    end
+    false
+  end
+
+  def my_any?(*arguments)
+    if !arguments[0].nil?
+      my_each { |index| return false unless arguments[0] == index }
+    elsif !block_given?
+      my_each { |index| return false unless index }
+    elsif arguments[0].is_a? Class
+      my_each { |index| return false unless index.class.ancestors.include?(arguments[0]) }
+    elsif arguments[0].is_a? Regexp
+      my_each { |index| return false unless arguments[0].match(index) }
+    else
+      my_each { |index| return false unless yield(index) }
+    end
+    true
+  end
+
```

Comment:
### `my_any?` should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
Refer to previous tests above 👆

Reply:
[Result Screenshot](https://imgur.com/a/ple2Dr0)